pkg/csi/service/osutils/windows_os_utils.go

```diff
@@ -511,3 +511,8 @@ func (osUtils *OsUtils) ShouldContinue(ctx context.Context) {
 	}
 	return
 }
+
+// Check if device at given path is block device or not
+func (osUtils *OsUtils) IsBlockDevice(ctx context.Context, volumePath string) (bool, error) {
```

Comment:
Have you tested this on Windows?  We are also going to GA Windows support so it should work there as well.

Reply:
As per CSI doc, NodeExpandVolume() API requires the volume to be published or staged on node. However, NodeStageVolume() and NodePublishVolume() APIs are blocked for raw block volumes in Windows vSphere CSI driver, with appropriate error returned. Hence, we do not expect NodeExpandVolume() getting called in case of Windows CSI raw block volumes.

Reply:
Tested this behavior on Windows setup, where the Pod remained stuck in ContainerCreating state due to NodeStageVolume() returning error and hence during online expansion, NodeExpandVolume() was not called.

Reply:
So what is our support statement for raw block on Windows?  Is it not supported?

Reply:
Yes, we will document that we will not support raw block volumes in Windows CSI.